crates/ra_ide/src/hover.rs

```diff
@@ -335,6 +346,277 @@ fn hover_for_definition(db: &RootDatabase, def: Definition) -> Option<Markup> {
     }
 }
 
+// Rewrites a markdown document, resolving links using `callback` and additionally striping prefixes/suffixes on link titles.
+fn map_links<'e>(
+    events: impl Iterator<Item = Event<'e>>,
+    callback: impl Fn(&str, &str) -> (String, String),
+) -> impl Iterator<Item = Event<'e>> {
+    let mut in_link = false;
+    let mut link_target: Option<CowStr> = None;
+
+    events.map(move |evt| match evt {
+        Event::Start(Tag::Link(_link_type, ref target, _)) => {
+            in_link = true;
+            link_target = Some(target.clone());
+            evt
+        }
+        Event::End(Tag::Link(link_type, _target, _)) => {
+            in_link = false;
+            Event::End(Tag::Link(link_type, link_target.take().unwrap(), CowStr::Borrowed("")))
+        }
+        Event::Text(s) if in_link => {
+            let (link_target_s, link_name) = callback(&link_target.take().unwrap(), &s);
+            link_target = Some(CowStr::Boxed(link_target_s.into()));
+            Event::Text(CowStr::Boxed(link_name.into()))
+        }
+        Event::Code(s) if in_link => {
+            let (link_target_s, link_name) = callback(&link_target.take().unwrap(), &s);
+            link_target = Some(CowStr::Boxed(link_target_s.into()));
+            Event::Code(CowStr::Boxed(link_name.into()))
+        }
+        _ => evt,
+    })
+}
+
+/// Rewrite documentation links in markdown to point to an online host (e.g. docs.rs)
+fn rewrite_links(db: &RootDatabase, markdown: &str, definition: &Definition) -> String {
+    let doc = Parser::new_with_broken_link_callback(
+        markdown,
+        Options::empty(),
+        Some(&|label, _| Some((/*url*/ label.to_string(), /*title*/ label.to_string()))),
+    );
```

Comment:
I think we'll eventually need to pull a full-fledged markdown parser, but I think for the time being it would be easier to just do some simple link extraction by hand? Like, finding `[non_ws*]` or something like that.

Reply:
Unfortunately it's a little more complicated than that. For one, footnote links aren't especially simple to parse. Add to the fact that links can appear in code blocks which we don't want to rewrite...

It could be done, but it would be a decent amount of work to do it without sacrificing correctness.